### PR TITLE
sota-client-cpp: GENIVI SOTA project in C++

### PR DESCRIPTION
**Pull request by leon-anavi**
Add Yocto/OE recipe for building GENIVI SOTA
project written in C++ with Boost C++ libraries.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

### CODE REVIEW

File: recipes-sota/sota-client-cpp/sota-client-cpp_git.bb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+S = "${WORKDIR}/git"
+
+SRCREV = "7194ec4aa39f71157af66751098340c854537817"
+
+SRC_URI = " \
+	git://github.com/advancedtelematic/sota_client_cpp \
+	"
+
+DEPENDS = "boost curl openssh"
```
Comment (cajun-rat) [inline recipes-sota/sota-client-cpp/sota-client-cpp_git.bb]: Did you mean _openssl_ here?

Comment (leon-anavi) [inline recipes-sota/sota-client-cpp/sota-client-cpp_git.bb]: Yes, thank you for pointing out this bug. I will fix it right away.

Best regards,
Leon

**Comment by leon-anavi** [comment]
Hi @cajun-rat,

The first commit adds Yocto/OE recipe for building SOTA Client ++ and the second commit ensures that bitbake builds the binary in Release mode.

Best regards,
Leon

**Comment by cajun-rat** [comment]
Hi @leon-anavi 

I built this yesterday but got pulled away before I managed to test it. We've got a product launch tomorrow, so I will hold off merging this until Friday :)

**Comment by leon-anavi** [comment]
Hi @cajun-rat,

OK, take your time. In the mean time I will continue working on:
* systemd service
* building SOTA client C++ with -DBUILD_WITH_DBUS_GATEWAY=ON

Best regards,
Leon

**Comment by cajun-rat** [comment]
Merged as dd5686ad0dfa28ef4c391addf184b2a0f99eda54

Review Score: 11/11

**Pull request closed without merging**